### ISSUE TITLE
Cielo plugin doesn't appear to properly start up

### ISSUE THREAD

**Issue by fractically** [ryanfroese/homebridge-smartcielo#3]
**Describe Your Problem:**

I used a different plugin successfully in the past with the same AC unit and WiFi dongle. However, since rebuilding my Homebridge system/setup (which is based on Hoobs, running on a Raspberry Pi) and installing this plugin, my Mr. Cool device - which is connected to and functional on my home network and controllable via the app - is not showing up in Homebridge, nor does it appear in Apple Homekit. I've entered my environmental values as instructed by the documentation to no avail, and most notably there are numerous errors (text in red) and a warning (text in orange) in the console when the plugin initializes.

FWIW, I've been perplexed by the 'Accessory Port Pool' start and end port fields - I don't see any mention of them in the instructions, but should these be set to something?

And in case it's helpful, here are other system component versions:
npm: '8.19.4'
node: '16.20.1'
v8: '9.4.146.26-node.26'
uv: '1.43.0'
zlib: '1.2.11'
brotli: '1.0.9'
ares: '1.19.1'
modules: '93'
nghttp2: '1.47.0'
napi: '8'
llhttp: '6.0.11'
openssl: '1.1.1u+quic'
cldr: '41.0'
icu: '71.1'
tz: '2022f'
unicode: '14.0'
ngtcp2: '0.8.1'
nghttp3: '0.7.0'

Thanks for any help you can provide!

**Logs:**

```
7/14/2023, 6:41:00 PMCielo Bridge-MrCool starting
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolLoaded plugin 'homebridge-cielo'
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolLoading 1 platforms...
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolBridge is running on port 51856.
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolERRORTypeError: Cannot read properties of undefined (reading 'user')
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolERRORat /var/lib/hoobs/cielobridge-mrcool/node_modules/node-smartcielo-ws/Cielo.js:264:52
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolERRORat processTicksAndRejections (node:internal/process/task_queues:96:5)
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolERRORat CieloAPIConnection.#getAccessTokenAndSessionId (/var/lib/hoobs/cielobridge-mrcool/node_modules/node-smartcielo-ws/Cielo.js:260:23)
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolERRORat CieloAPIConnection.establishConnection (/var/lib/hoobs/cielobridge-mrcool/node_modules/node-smartcielo-ws/Cielo.js:119:5)
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolERRORat HomebridgeAPI.<anonymous> (/var/lib/hoobs/cielobridge-mrcool/node_modules/homebridge-cielo/src/platform.ts:73:7)
7/14/2023, 6:41:14 PMCielo Bridge-MrCoolWARNINGunhandled rejection: TypeError: Cannot read properties of undefined (reading 'sessionId')
```

**Plugin Config:**

```json
{
    "platform": "Smartcielo",
    "username": "xxxxx@provider.tld",
    "password": "xxxxx",
    "ip": "xx.xx.181.120",
    "macAddresses": [
        "500291e4d507"
    ]
}
```

**Screenshots:**

![pluginconfig](https://github.com/ryanfroese/homebridge-smartcielo/assets/74267652/422be933-0b81-4d6f-8515-8043e2bb60f1)

**Environment:**

* **Plugin Version**: 1.1.5
* **Homebridge Version**: 1.6.0
* **Node.js Version**: 16.20.1
* **NPM Version**: 8.19.4
* **Operating System**: Raspbian

**Comment by jardilio** [comment]
Any update? I see the same error, but it was working previously. I'm also running as a child bridge and it keeps restarting over and over.

**Comment by BMWAdriano** [comment]
> Any update? I see the same error, but it was working previously. I'm also running as a child bridge and it keeps restarting over and over.

I am having the exact same issue. I tried using the plugin as a child bridge and it continuously restarts. Without using as a child bridge, it crashes the Homebridge completely. Help?

**Comment by fractically** [comment]
*bump*

At this point I'd be happy with "leave me alone, I have no idea what's going wrong", "I don't have time to continue working on this", or even a simple "NO", and I'll give up and move onto another solution. I get it, no one is making money off of whether this works for end users, but it would be great if the person who built this project would say *something*... Many of us are having the exact same issue, so it's not just a single, crazy-weird environmental issue...

**Comment by SkyJetty** [comment]
Tried to get this working and it just crashes and forces home bridge to reboot over and over just like all the other versions I have tried.

**Comment by rdjfraser** [comment]
@fractically looks like a lot more discussion here on issues caused by frequent changes to API causing breaks. I don't see anyone actively working on it :( https://github.com/isaac-webb/homebridge-mrcool/issues/3

**Comment by salmundo11** [comment]
Same issue here, have installed and configured the plugin and it causes a reboot loop.

**Comment by fractically** [comment]
> [@fractically](https://github.com/fractically) looks like a lot more discussion here on issues caused by frequent changes to API causing breaks. I don't see anyone actively working on it :( [isaac-webb#3](https://github.com/isaac-webb/homebridge-mrcool/issues/3)

Yeah, it looks like you're right. Not sure how I missed all of the captcha and API chatter that started up in 2023. I came here tonight because I got a cryptic email notification about a cryptic reply (or at least this 'something' appeared in the conversation:
https://github.com/ryanfroese/homebridge-smartcielo/issues/5
I'm honestly not sure if this is a reply to something I posted, or even what it means:
https://github.com/ryanfroese/homebridge-smartcielo/issues/5#event-19149730635

The post simply says:
**aadittmar** closed this as [not planned](https://github.com/ryanfroese/homebridge-smartcielo/issues?q=is%3Aissue%20state%3Aclosed%20archived%3Afalse%20reason%3Anot-planned)[2 hours ago](https://github.com/ryanfroese/homebridge-smartcielo/issues/5#event-19149730635)

Anyone care to say (with confidence) whether "this" implies the ENTIRE project is dead? I don't know what else could be meant by the statement, since no solution was offered, meaning the plugin is still DOA.

**Comment by ryanfroese** [comment]
Hi, I fixed this FYI. I just did it now, but had to add a captcha solving service to get mine up and running. This should now work if you want to give it a shot?
